_rules/element-lang-programmatically-determinable-7ed469.md

```diff
@@ -0,0 +1,142 @@
+---
+id: 7ed469
+name: Element language is programmatically determinable
+rule_type: atomic
+description: |
+  This rule checks that the language of an element in the page body can be programmatically determined.
+accessibility_requirements: # Remove whatever is not applicable
```

Comment:
I think the rule is also mapping to 3.1.1: if an element has no ancestor with a valid `lang`, then the document element has no valid `lang`.

Reply:
i understood `accessibility_requirements` as the list of (in this case) WCAG SCs that this is used to test conformance to. as this rule is scoped to descendants of `body`, it cannot be used to test 3.1.1 language of the page itself, as that needs to (in HTML) be defined for the `html` element?

Reply:
I agree that the rule is not designed to test SC 3.1.1, but it still does test a bit of it indirectly…
As always, we focus on the fail-on-fail relationship. A failure of this rule indicates a failure of SC 3.1.1 (in addition to 3.1.2). A Pass on this rule does not indicate a Pass of 3.1.1, but neither of 3.1.2… That's how it is for most rules.

Then, it is true also that a failure of this rule implies a failure of [HTML page has lang attribute](https://act-rules.github.io/rules/b5c3f8), and that is why it fails 3.1.1. That's also OK. This rule checks more.

Reply:
wondering now if another failure purely for this rule (which i've not added here, but can if it's deemed an actual failure) would be something like
```
<html lang="en">
  <body>
    <div lang="foo">
      <p>Content</p>
    </div>
  </body>
</html>
```

where the `p` fails because its ancestor `div` has an invalid language attribute, *but* the page itself does pass 3.1.1...and so a failure of this rule does not indicate a failure of 3.1.1

Reply:
The rule currently does not fail that because "any of its [the `p`] ancestor" has a valid `lang` tag.
I agree that this is a failure of 3.1.2 but not of 3.1.1

I see 2 possibilities:
=> Change the rule to "its closest ancestor with a `lang` attribute" or something similar. This makes the rule more tricky. In this case, it would indeed not map to 3.1.1
=> Leave the rule as is. This keeps the rule simpler and more focused. This case is already handled by [Element within `body` has valid `lang` attribute](https://act-rules.github.io/rules/de46e4), so there is no real need to enforce it in this rule. The main "problem" with this solution is that Passed Examples must pass the SC, so we cannot use this as a Passed Example to illustrate the limitation of the rule. This is not really a big deal…

Given that this case is already handle by anther rule, I'd rather keep this rule simple.